lib/azure/armrest/resource_group_based_service.rb

```diff
@@ -46,7 +46,16 @@ def list(rgroup = configuration.resource_group)
         url = yield(url) || url if block_given?
         response = rest_get(url)
 
-        Azure::Armrest::ArmrestCollection.create_from_response(response, model_class)
+        results = Azure::Armrest::ArmrestCollection.create_from_response(response, model_class)
+
+        while results.continuation_url
+          response = rest_get_without_encoding(results.continuation_url)
+          more = Azure::Armrest::ArmrestCollection.create_from_response(response, model_class)
+          results.concat(more)
+          results.continuation_url = more.continuation_url
+        end
+
+        results
       end
```

Comment:
@djberg96 - I would have created a separate local variable called eg ```continuation_url``` and set/updated it as needed; it feels uncomfortable to me to overwrite the url in the original ```results``` array. This is only a minor style comment though, feel free to ignore or take on board.